File: scripts/io.py
from helpers import read_file, file_to_grange, init_ray; init_ray(snakemake.wildcards)

from time import time
import datetime

operation = snakemake.wildcards.io_operation
w = snakemake.wildcards

f = snakemake.input[0]

print(snakemake.params.code)
start = time()
exec(snakemake.params.code)
end = time()
total = end - start

print(result)

total_dt = datetime.datetime.fromtimestamp(total)

minutes_seconds = total_dt.strftime('%-M.%-S.%f')

open(snakemake.output.time, "w+").write(minutes_seconds)

result_string = str(result)

open(snakemake.output.result, "w+").write(result_string)

File: scripts/io.R
source("scripts/helpers.R")

library(rtracklayer)

operation = snakemake@params[["io_operation"]]
operation = paste0("f <- function(gr){\n", operation, "\nreturn(result)}")
print(paste0("Performing operation ", operation, " for ", snakemake@wildcards[["unary_operation"]]))

file = snakemake@input[[1]]

eval(parse(text=operation))
start.time <- Sys.time()

result = f(gr)

end.time <- Sys.time()

time.taken <- end.time - start.time

time.taken <- as.numeric(time.taken, units="secs")

write(time.taken, snakemake@output[["time"]])

result_string = capture.output(print(result))
write(result_string, snakemake@output[["result"]])
